### ISSUE TITLE
grpo can not support deepseek-6.7b-base model

### ISSUE THREAD

**Issue by aacedar** [modelscope/ms-swift#4785]
使用qwen模型可以跑grpo，但替换成deepseek模型无法运行，报错如下：

<img width="1258" alt="Image" src="https://github.com/user-attachments/assets/893a44cc-91f1-47a8-a6de-6740b48a6279" />

将model参数从 ‘Qwen/Qwen2.5-7B-Instruct’ 替换成 ‘deepseek-ai/deepseek-coder-6.7b-base’就报如上图错误
完整的启动脚本
`
CUDA_VISIBLE_DEVICES=0,1,2,3,4,5 \
NPROC_PER_NODE=6 \
swift rlhf \
    --rlhf_type grpo \
    --model ./deepseek-ai/deepseek-coder-6.7b-base \
    --reward_funcs accuracy \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --train_type full \
    --torch_dtype bfloat16 \
    --dataset AI-MO/NuminaMath-TIR#1000 \
    --split_dataset_ratio 0 \
    --max_completion_length 2048 \
    --num_train_epochs 3 \
    --per_device_train_batch_size 2 \
    --learning_rate 1e-6 \
    --gradient_accumulation_steps 2 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --warmup_ratio 0.05 \
    --dataloader_num_workers 4 \
    --dataset_num_proc 4 \
    --num_generations 6 \
    --temperature 0.9 \
    --top_p 0.9 \
    --top_k 50 \
    --deepspeed zero2 \
    --log_completions true \
    --num_iterations 1 \
    --beta 0.04 \
    --output_dir output \
    --report_to tensorboard
`

swift.__version__: 3.5.0

**Comment by hjh0119** [comment]
fixed in https://github.com/modelscope/ms-swift/pull/4827

**Comment by aacedar** [comment]
> fixed in [#4827](https://github.com/modelscope/ms-swift/pull/4827)

没有解决彻底，这里的逻辑是将非str的信息过滤掉，但这些special token在templateMeta的基类中都转成了token_id， 按这个做法，就默认将这些special token丢弃了，这个做法虽然可以运行但不符合实际情况， 解决方案见：https://github.com/modelscope/ms-swift/pull/4833
```
prompts_text.append(''.join(elem for elem in res_context_list if isinstance(elem, str)))

```
